src/node/desktop/CMakeLists.txt

```diff
@@ -16,35 +16,39 @@
 cmake_minimum_required(VERSION 3.4.3)
 project(ELECTRON_DESKTOP)
 
+# install mac-terminal script on apple
+if(APPLE)
+   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/mac-terminal.in
+                  ${CMAKE_CURRENT_BINARY_DIR}/mac-terminal)
+endif()
+
 # defines NODEJS, NODEJS_PATH, YARN, YARN_PATH
 include("${CMAKE_CURRENT_LIST_DIR}/../CMakeNodeTools.txt")
 
 # ensure yarn and node are on the path
 set(MODIFIED_PATH "${NODEJS_PATH}:${YARN_PATH}:$ENV{PATH}")
 
-add_custom_target(electron_build ALL)
-if (TARGET gwt_build)
-   add_dependencies(electron_build gwt_build rsession diagnostics)
-else()
-   # for iterating on build infrastructure, skipping gwt build
-   add_dependencies(electron_build rsession diagnostics)
-endif()
+# copy sources to build directory. note that the build directory cannot
+# be the "true" CMake directory as some files are resolved relative to
+# the desktop project's relative path in the application structure
+set(ELECTRON_BUILD_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../desktop-build-${UNAME_M}")
+file(REMOVE_RECURSE "${ELECTRON_BUILD_DIR}")
+file(MAKE_DIRECTORY "${ELECTRON_BUILD_DIR}")
+file(
+	COPY "${CMAKE_CURRENT_SOURCE_DIR}/"
+   DESTINATION "${ELECTRON_BUILD_DIR}/"
+	REGEX "/out$" EXCLUDE
+	REGEX "/node_modules$" EXCLUDE)
 
-# install mac-terminal script on apple
-if(APPLE)
-   configure_file(${CMAKE_CURRENT_SOURCE_DIR}/mac-terminal.in
-                  ${CMAKE_CURRENT_BINARY_DIR}/mac-terminal)
+# build electron components
+add_custom_target(
+   electron_build ALL
+   DEPENDS rsession diagnostics
+   WORKING_DIRECTORY "${ELECTRON_BUILD_DIR}"
+   COMMAND ${CMAKE_COMMAND} -E env "PATH=${MODIFIED_PATH}" "PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=1" ${YARN} make
```

Comment:
We probably want `yarn pkg` or something else that avoids the package-building step (as we haven't yet copied in the session components required at packaging time).

Reply:
We can just remove the DMG and signing steps from `yarn make`. do you want me to do it?

Reply:
Sounds good!